src/gt4py/next/ffront/foast_to_gtir.py

```diff
@@ -282,9 +291,11 @@ def _visit_shift(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
         for arg in node.args:
             match arg:
                 # `field(Off[idx])`
-                case foast.Subscript(value=foast.Name(id=offset_name), index=int(offset_index)):
+                case foast.Subscript(value=foast.Name(id=offset_name), index=index):
+                    index = constant_folding.ConstantFolding().visit(self.visit(index, **kwargs))
```

Comment:
```suggestion
                    index = constant_folding.ConstantFolding.apply(self.visit(index, **kwargs))
```
Could you leave a comment why this is done here? A dynamic expression would also work right?